core/src/main/java/nu/yona/server/subscriptions/service/UserAddService.java

```diff
@@ -230,20 +233,21 @@ private void assertWhiteList(String mobileNumber, UserSignUp origin)
 
 	private void deleteExistingUserToOverwriteIt(String mobileNumber, String userProvidedConfirmationCode)
 	{
-		User existingUserEntity = userLookupService
-				.lockUserForUpdate(UserLookupService.findUserByMobileNumber(mobileNumber).getId());
+		User existingUserEntity = userLookupService.lockUserForUpdate(
+				UserLookupService.findUserByMobileNumber(mobileNumber).getId());
 		Optional<ConfirmationCode> confirmationCode = existingUserEntity.getOverwriteUserConfirmationCode();
 
 		assertValidConfirmationCode(existingUserEntity, confirmationCode, userProvidedConfirmationCode,
 				() -> UserOverwriteConfirmationException.confirmationCodeNotSet(existingUserEntity.getMobileNumber()),
-				r -> UserOverwriteConfirmationException
-						.confirmationCodeMismatch(existingUserEntity.getMobileNumber(), userProvidedConfirmationCode, r),
+				r -> UserOverwriteConfirmationException.confirmationCodeMismatch(existingUserEntity.getMobileNumber(),
+						userProvidedConfirmationCode, r),
 				() -> UserOverwriteConfirmationException.tooManyAttempts(existingUserEntity.getMobileNumber()));
 
 		// notice we can't delete the associated anonymized data
 		// because the anonymized data cannot be retrieved
 		// (the relation is encrypted, the password is not available)
-		logger.info("DEBUG: mobile number {} delete/overwrite user with confirmation code ID {}", mobileNumber, confirmationCode.get().getId());
+		logger.info("DEBUG: mobile number {} delete/overwrite user with confirmation code ID {}", mobileNumber,
+				confirmationCode.get().getId());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "confirmationCode.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS3655)